wlib/LoginDlg.cpp

```diff
@@ -934,16 +935,43 @@ bool PopulateConfigCombo(CComboBox &configCombo, const std::_tstring & defaultVa
 			}
 		}
 	}
-	if(configCombo.GetCount() == 0)
+	if (configCombo.GetCount() == 0)
 	{
 		configCombo.AddString(_T("default"));
 		retVal = false;
 	}
-	if ( matchID < 0 )
+	if (matchID < 0)
 		configCombo.SetCurSel(0);
 	return retVal;
 }
 
+bool PopulateCTConfigCombo(CComboBox & configCTCombo, const std::_tstring & defaultValue)
+{
+	bool retVal = true;
+	EnableLocalRepository(TRI_BOOL_TRUE);
+	if (CComPtr<IEclCC> eclcc = CreateIEclCC(true)) {
+		configCTCombo.ResetContent();
+		std::vector<std::wstring> compilers;
+		std::wstring currentVersion = GetCompilers(compilers);
+		int i = 0;
+		int current = 0;
+		for (auto& c : compilers)
+		{
+			if (defaultValue.length()) {
+				if (_tcsicmp(c.c_str(), defaultValue.c_str()) == 0)
+					current = i;
+			}
+			else if (_tcsicmp(c.c_str(), currentVersion.c_str()) == 0)
+				current = i;
+			configCTCombo.AddString(c.c_str());
```

Comment:
It might make life easier to iterator over the `g_foundCompilers`:
```
int item = configCTCombo.AddString(c.GetString());
configCTCombo.SetItemData(item, (DWORD_PTR)c);
```

Reply:
@GordonSmith will take a look.

Reply:
@GordonSmith you can't make vectors of abstract classes and I have tried numerous ways to do this. I think we should leave it as is for now.